File: app/src/main/java/com/sayler/gina/store/settings/SPSettingsStoreManager.java
package com.sayler.gina.store.settings;

import android.content.Context;
import com.sayler.gina.store.SharedPreferencesHelper;
import com.sayler.gina.util.Constants;

import javax.inject.Singleton;

@Singleton
public class SPSettingsStoreManager implements SettingsStoreManager {
  private final SharedPreferencesHelper<SettingsStore> sharedPreferencesHelper;

  public SPSettingsStoreManager(final Context context) {
    sharedPreferencesHelper = new SharedPreferencesHelper<>(context, Constants.PREF_SETTINGS_STORE, SettingsStore.class);
  }

  @Override
  public SettingsStore get() {
    return sharedPreferencesHelper.retrieve();
  }

  @Override
  public void save(SettingsStore object) {
    sharedPreferencesHelper.store(object);
  }

  @Override
  public void clear() {
    sharedPreferencesHelper.clear();
  }

}

File: app/src/main/java/com/sayler/gina/store/settings/SPSettingsStoreManager.kt
package com.sayler.gina.store.settings

import android.content.Context
import com.sayler.gina.store.SharedPreferenceHelper
import com.sayler.gina.util.Constants

import javax.inject.Singleton

@Singleton
class SPSettingsStoreManager(context: Context) : SettingsStoreManager {
    var sharedPreferenceHelper: SharedPreferenceHelper<SettingsStore> = SharedPreferenceHelper(context, Constants.PREF_SETTINGS_STORE, SettingsStore::class.java)

    override fun get(): SettingsStore? {
        return sharedPreferenceHelper.retrieve()
    }

    override fun save(`object`: SettingsStore) {
        sharedPreferenceHelper.store(`object`)
    }

    override fun clear() {
        sharedPreferenceHelper.clear()
    }

}